internal/redfishwrapper/virtual_media.go

```diff
@@ -29,50 +33,43 @@ func (c *Client) SetVirtualMedia(ctx context.Context, kind string, mediaURL stri
 		return false, errors.New("invalid media type")
 	}
 
-	for _, manager := range managers {
-		virtualMedia, err := manager.VirtualMedia()
+	for _, m := range managers {
+		virtualMedia, err := m.VirtualMedia()
 		if err != nil {
 			return false, err
 		}
-		for _, media := range virtualMedia {
-			if media.Inserted {
-				err = media.EjectMedia()
-				if err != nil {
-					return false, err
-				}
-			}
+		if len(virtualMedia) == 0 {
+			return false, errors.New("no virtual media found")
 		}
-	}
 
-	// An empty mediaURL means eject everything, so if that's the case we're done. Otherwise, we
-	// need to insert the media.
-	if mediaURL != "" {
-		setMedia := false
-		for _, manager := range managers {
-			virtualMedia, err := manager.VirtualMedia()
-			if err != nil {
-				return false, err
+		for _, vm := range virtualMedia {
+			if vm.Inserted {
+				if err := vm.EjectMedia(); err != nil {
+					return false, err
+				}
 			}
-
-			for _, media := range virtualMedia {
-				for _, t := range media.MediaTypes {
-					if t == mediaKind {
-						err = media.InsertMedia(mediaURL, true, true)
-						if err != nil {
+			if mediaURL != "" {
```

Comment:
could this return early by inverting the check? that would make it easier to read as well